subsys/bluetooth/host/hci_core.c

```diff
@@ -3133,6 +3133,34 @@ static void read_buffer_size_complete(struct net_buf *buf)
 #endif /* !defined(CONFIG_BT_CLASSIC) */
 #endif /* CONFIG_BT_CONN */
 
+#if defined(CONFIG_BT_CONN) || defined(CONFIG_BT_ISO)
+static void check_num_of_acl_iso_pkts(uint8_t num_acl_pkts, uint8_t num_iso_pkts)
+{
+	/** This check is added to warn if the number of ACL or ISO packets in the controller is not
+	 * equal to the number of bt_conn_tx contexts allocated by the host. The inequality of these
+	 * two values can lead to inefficient resources usage either on the host's or controller's
+	 * side.
+	 */
+#if defined(CONFIG_BT_CONN)
+	CHECKIF(bt_conn_tx_acl_count() != num_acl_pkts) {
+		LOG_WRN("Num of Controller's ACL packets != ACL bt_conn_tx contexts (%u != %u)",
+			num_acl_pkts, bt_conn_tx_acl_count());
+	}
+#else
+	ARG_UNUSED(num_acl_pkts);
+#endif
+
+#if defined(CONFIG_BT_ISO)
+	CHECKIF(bt_conn_tx_iso_count() != num_iso_pkts) {
```

Comment:
What is the rationale of using CHECKIF here, compared to `if` or assert?

Reply:
The idea is that you can remove this check (and the warning itself) with `CONFIG_CONFIG_NO_RUNTIME_CHECKS`.

Reply:
Right. We discussed the use of CHECKIF in a (fairly) recent meeting in the BT WG (https://github.com/zephyrproject-rtos/zephyr/issues/31341). 
I generally like the idea, but one issue with it is that lack of control; either all of them are removed, are `if`s or are asserts. 
So if the application wants to assert of e.g. incorrect API usage and sets `CONFIG_ASSERT_ON_ERRORS=y`, then this warning becomes an assert instead, which may be is not what we want as it's not really breaking anything (is it)?

Reply:
Changed to `if`.